Cannot find module 'sanity:client' or its corresponding type declarations.

**Describe the bug**

I am following the instructions in [this article](https://www.sanity.io/plugins/sanity-astro) to build an astro project, but an error occurred during the build process.

This bug occurs during the build process and does not appear during the dev process.

Is this something I can solve on my own now? Or do I have to wait for you to fix it?

Thanks.

**To Reproduce**

Dependencies:

    "@astrojs/check": "^0.3.4",
    "@astrojs/react": "^3.0.9",
    "@sanity/astro": "^2.2.0",
    "@sanity/client": "^6.10.0",
    "@types/react": "^18.2.46",
    "@types/react-dom": "^18.2.18",
    "astro": "^4.1.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "sanity": "^3.23.4",
    "typescript": "^5.3.3"

astro.config.mjs:
```
import { defineConfig } from 'astro/config';
import { sanityIntegration } from "@sanity/astro";

export default defineConfig({
  integrations: [sanityIntegration({
    projectId: '***',
    dataset: '***',
    apiVersion: '2022-03-07',
    useCdn: false
  })],
  image: {
    domains: ["cdn.sanity.io"],
    remotePatterns: [{
      protocol: "https"
    }]
  }
});

```

Try to fetch data inside a component:

```
---
import { sanityClient } from "sanity:client";

export async function getStaticPaths() {
  const slugs = await sanityClient.fetch(`*[_type == 'book']{
  "slug": slug.current,
  "title": title
}`);
  const paths = await slugs.map((book: Book) => {
    return {
      params: {
        book: book.slug,
      },
      props: {title: book.title}
    };
  });
  return paths;
}
```

run `npm run build` will cause this error: 

 `error ts(2307): Cannot find module 'sanity:client' or its corresponding type declarations.`

**Expected behavior**

should build successfully

**Which versions of Sanity are you using?**

```
@sanity/cli (global)  3.23.4 (up to date)
@sanity/astro          2.2.0 (up to date)
sanity                3.23.4 (up to date)
```

**What operating system are you using?**

macOS 14.2.1

**Which versions of Node.js / npm are you running?**

18.18.0/10.1.0

--- a/apps/example-ssr/src/env.d.ts
+++ b/apps/example-ssr/src/env.d.ts
@@ -1,1 +1,2 @@
 /// <reference types="astro/client" />
+/// <reference types="@sanity/astro/module" />

--- a/apps/example/src/env.d.ts
+++ b/apps/example/src/env.d.ts
@@ -1,1 +1,2 @@
 /// <reference types="astro/client" />
+/// <reference types="@sanity/astro/module" />

--- a/packages/sanity-astro/package.json
+++ b/packages/sanity-astro/package.json
@@ -19,6 +19,7 @@
       "module": "./dist/sanity-astro.mjs",
       "default": "./dist/sanity-astro.mjs"
     },
+    "./module": "./module.d.ts",
     "./studio/studio-route.astro": "./dist/studio/studio-route.astro",
     "./studio/studio-component.tsx": {
       "types": "./src/studio/studio-component.tsx",

